### PR TITLE
Introduce wc_attribute_taxonomy_slug()

### LINKED ISSUE #22101: Attributes containing "pa_" are not managed properly
**Describe the bug**
If an attribute contains "pa_" in its slug, it is not managed correctly. The cause is that WooCommerce replaces all occurrences of "pa_" instead of replacing only if the taxonomy starts with "pa_".
In the code: str_replace('pa_', '') is used instead of preg_replace('/^pa_/', '').

One consequence is that the attribute name is wrong. For example, the attribute "SPA test" is displayed as "stest" in the backend and in the frontend.

**To Reproduce**
1. Create an attribute named "SPA test" with the slug "spa_test" (with an underscore)
2. Create a variable product - Go to Attributes section
3. Add the "SPA test" attribute
4. The attribute name displayed is "stest" instead of "SPA test" (backend and frontend)

**Screenshots**
See the attached 3 screenshots.

**Expected behavior**
WooCommerce should not remove "pa_" in the attribute name.

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
WordPress 4.9.8
WooCommerce 3.5.2
```
</details>
test product test import
edit product test import wordpress 2
attributes test import wordpress

### PR DESCRIPTION
**Pull request by rrennick**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

- Fix phpcs sniffs in all touched files
- Standardize stripping of `pa_` prefix from product attribute taxonomies by introducing `wc_attribute_taxonomy_slug()`

Closes #22101 .

### How to test the changes in this Pull Request:

1. Create a Product Attribute containing `pa_` in the slug. 
2. Add a couple terms to the taxonomy.
3. Without this PR add the attribute to a product. The attribute edit is the local attribute edit.
4. With this PR add the attribute to a product. The attribute edit is the global attribute edit. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### Changelog entry

> Fix, allow `pa_` in product attribute slugs.

### CODE REVIEW

**Review by mikejolley — COMMENTED** [review]
Looks good overall @rrennick 

Was _raw() the best name for this? I cannot think of a better one but I'm not sure how much sense it makes.

Could you update the `@since` tag to 3.6. please?

**Comment by rrennick** [comment]
@mikejolley How do `wc_attribute_taxonomy_slug()` or  `wc_attribute_real_taxonomy_name()` sound?

**Comment by mikejolley** [comment]
Perhaps `slug` yes.

Would that inger that `wc_attribute_taxonomy_name` should be a 'name' rather than the taxonomy?`wc_attribute_taxonomy_name` returns something like pa_color. Could get confusing, that other function is not named well.

**Comment by rrennick** [comment]
> Would that infer that wc_attribute_taxonomy_name should be a 'name' rather than the taxonomy?

In the DB, the taxonomy name has the `pa_` while in the interface it doesn't so both are taxonomy names. `slug` is used in existing code handling terms so we won't be introducing new terminology.

**Comment by rrennick** [comment]
@mikejolley Updated per feedback. Thanks

File: includes/admin/class-wc-admin-importers.php [hunk 1]
```diff
@@ -159,7 +159,7 @@ public function post_importer_compatibility() {
 					foreach ( $post['terms'] as $term ) {
 						if ( strstr( $term['domain'], 'pa_' ) ) {
 							if ( ! taxonomy_exists( $term['domain'] ) ) {
-								$attribute_name = wc_sanitize_taxonomy_name( str_replace( 'pa_', '', $term['domain'] ) );
```
Comment (mikejolley) [inline includes/admin/class-wc-admin-importers.php]: Is this an err? Looks like the pa version was used here?

File: includes/wc-attribute-functions.php [hunk 2]
```diff
@@ -131,7 +132,7 @@ function wc_attribute_taxonomy_id_by_name( $name ) {
  */
 function wc_attribute_label( $name, $product = '' ) {
 	if ( taxonomy_is_product_attribute( $name ) ) {
-		$name       = wc_sanitize_taxonomy_name( str_replace( 'pa_', '', $name ) );
```
Comment (mikejolley) [inline includes/wc-attribute-functions.php]: another name vs slug

File: includes/widgets/class-wc-widget-layered-nav.php [hunk 3]
```diff
@@ -452,7 +452,7 @@ protected function layered_nav_list( $terms, $taxonomy, $query_type ) {
 
 				// Add Query type Arg to URL.
 				if ( 'or' === $query_type && ! ( 1 === count( $current_filter ) && $option_is_set ) ) {
-					$link = add_query_arg( 'query_type_' . sanitize_title( str_replace( 'pa_', '', $taxonomy ) ), 'or', $link );
```
Comment (mikejolley) [inline includes/widgets/class-wc-widget-layered-nav.php]: Can you see a reason `sanitize_title` was used here and in the layered nav filter widger previously? Possibly special chars?

**Review by mikejolley — COMMENTED** [review]
There are a few places where name was replaced with slug and I'm not sure if that was a mistake or not?

File: includes/admin/class-wc-admin-importers.php (hunk 1 above)
Comment (rrennick) [reply includes/admin/class-wc-admin-importers.php]: The new function uses `wc_sanitize_taxonomy_name` before doing the str_replace so the only difference is the processing order change.

https://github.com/woocommerce/woocommerce/pull/22112/files#diff-dc3a1c9d68e161cfe6566b05971ec631R681

File: includes/widgets/class-wc-widget-layered-nav.php (hunk 3 above)
Comment (rrennick) [reply includes/widgets/class-wc-widget-layered-nav.php]: The only reason I can see is that attributes/values can be the local ones and it was added as a precaution.

**Review by mikejolley — APPROVED** [review]

**Pull request merged**